errors/errors.go

```diff
@@ -113,6 +113,20 @@ func (e *Error) AddDetails(details ...proto.Message) *Error {
 	return e
 }
 
+// FromError takes in an error and returns back the kitError if it's that type under the hood
+func FromError(err error) (*Error, bool) {
+	if err == nil {
+		return nil, false
+	}
+
+	kitErr, ok := err.(Error)
```

Comment:
Use `errors.As` here, which supports the case where `Error` is wrapped inside another error.

Reply:
good idea, done.